src/injectors.ts

```diff
@@ -15,6 +29,13 @@ interface AmpExp {
   }>;
 }
 
+/**
+ * Compresses a text (assumed to be in UTF-8) to Brotli level 11.
+ */
+async function brotliCompress(text: string): Promise<Uint8Array> {
+  return brotli.compress(textEncoder.encode(text), {quality: 11});
+}
```

Comment:
I think a next step here is to move the brotli compression logic to a new file (`compress.ts`) that will do:
1. client supports Brotli? (no: return as-is; yes: continue)
2. check if we already compressed this file and cached it (yes: serve from cache; no: continue)
3. compress, cache, and serve

Reply:
From the discussion with CloudFlare, there's some post-request API that we can take advantage of that will allow us to queue compression after the first request, so that we never block. When compression is done, we can begin serving it out of cache. We just need to ensure we don't queue multiple compressions.

Reply:
I'm gonna guess this is what they were talking about: https://developers.cloudflare.com/workers/runtime-apis/fetch-event#waituntil

I'll try to think of how to structure this

Reply:
Check out the updated code. From a few quick local tests it looks like it does the job right (i.e., my first request to https://empty-butterfly-fee6.rodaniel-test1.workers.dev/v0/amp-geo-0.1.mjs was 5.3 kb, subsequent ones were 4.8 kb)

I think the cache is extremely short lived (~minutes) and it also drops the `cache-control` response header when serving cached dynamic injections, two things I'll look into tomorrow

Reply:
> I think the cache is extremely short lived (~minutes) and it also drops the `cache-control` response header when serving cached dynamic injections, two things I'll look into tomorrow

Fixed this